Make CleverCSV a pre-commit hook

Would you consider making CleverCSV a [pre-commit](https://pre-commit.com) hook, by adding a [`.pre-commit-hooks.yaml`](https://pre-commit.com/#creating-new-hooks) file to your repository? This would allow people to have csv files in their repositories to automatically check them whenever they want to commit them.

The only thing missing to make it work is, as I understand:
* An option to overwrite the input file instead of printing to STDOUT (I think this could be a flag, like `--in-place`)
* The possibility of `clevercsv` to accept a list of file arguments to process.
* (Probably implemented already) A different exit status whether the input file was altered (!=0) or not (=0).

I can file a PR for the `.pre-commit-hooks.yaml`, if you agree (after the above points are implemented).

Thanks for your consideration!

I have created [a fork](https://github.com/lcnittl/CleverCSV/tree/patch-1) and added the `.pre-commit-hooks.yaml` file there. However, I am having problems installing CleverCSV with pip and the git+https URI scheme (To my understanding, this is the way pre-commit would install the package too):

```
(.venv) PS C:\Users\lcnittl\playground> pip install --upgrade git+https://github.com/lcnittl/CleverCSV@patch-1
Collecting git+https://github.com/lcnittl/CleverCSV@patch-1
  Cloning https://github.com/lcnittl/CleverCSV (to revision patch-1) to c:\users\lcnittl\appdata\local\temp\pip-req-build-j5a594ud
Collecting chardet>=3.0
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting regex>=2018.11
  Using cached regex-2020.10.15-cp38-cp38-win_amd64.whl (268 kB)
Building wheels for collected packages: clevercsv
  Building wheel for clevercsv (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\lcnittl\playground\.venv\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\lcnittl\\AppData\\Local\\Temp\\pip-req-build-j5a594ud\\setup.py'"'"'; __file__='"'"'C:\\Users\\lcnittl\\AppData\\Local\\Temp\\pip-req-build-j5a594ud\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\lcnittl\AppData\Local\Temp\pip-wheel-p580azgi'
       cwd: C:\Users\lcnittl\AppData\Local\Temp\pip-req-build-j5a594ud\
  Complete output (70 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.8
  creating build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\break_ties.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\consistency.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\cparser_util.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\detect.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\detect_pattern.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\detect_type.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\dialect.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\dict_read_write.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\escape.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\exceptions.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\normal_form.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\potential_dialects.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\read.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\utils.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\wrappers.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\write.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\_optional.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\__init__.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\__main__.py -> build\lib.win-amd64-3.8\clevercsv
  copying clevercsv\__version__.py -> build\lib.win-amd64-3.8\clevercsv
  creating build\lib.win-amd64-3.8\clevercsv\console
  copying clevercsv\console\application.py -> build\lib.win-amd64-3.8\clevercsv\console
  copying clevercsv\console\config.py -> build\lib.win-amd64-3.8\clevercsv\console
  copying clevercsv\console\__init__.py -> build\lib.win-amd64-3.8\clevercsv\console
  creating build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\code.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\detect.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\explore.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\standardize.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\view.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\_utils.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  copying clevercsv\console\commands\__init__.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
  running egg_info
  creating clevercsv.egg-info
  writing clevercsv.egg-info\PKG-INFO
  writing dependency_links to clevercsv.egg-info\dependency_links.txt
  writing entry points to clevercsv.egg-info\entry_points.txt
  writing requirements to clevercsv.egg-info\requires.txt
  writing top-level names to clevercsv.egg-info\top_level.txt
  writing manifest file 'clevercsv.egg-info\SOURCES.txt'
  reading manifest file 'clevercsv.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'requirements.txt'
  warning: no files found matching '*' under directory 'bin'
  no previously-included directories found matching 'tests\test_integration'
  warning: no previously-included files found matching 'Makefile'
  warning: no previously-included files found matching '.gitignore'
  warning: no previously-included files found matching '.travis.yml'
  warning: no previously-included files found matching '.readthedocs.yml'
  warning: no previously-included files found matching 'make_release.py'
  warning: no previously-included files found matching 'cgrep'
  warning: no previously-included files found matching 'vgrep'
  no previously-included directories found matching 'notes'
  no previously-included directories found matching 'auxiliary'
  writing manifest file 'clevercsv.egg-info\SOURCES.txt'
  running build_ext
  building 'clevercsv.cparser' extension
  creating build\temp.win-amd64-3.8
  creating build\temp.win-amd64-3.8\Release
  creating build\temp.win-amd64-3.8\Release\src
  C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\lcnittl\playground\.venv\include -Ic:\python\python38\include -Ic:\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" /Tcsrc/cparser.c /Fobuild\temp.win-amd64-3.8\Release\src/cparser.obj
  cparser.c
  c:\python\python38\include\pyconfig.h(206): fatal error C1083: Cannot open include file: 'basetsd.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for clevercsv
  Running setup.py clean for clevercsv
Failed to build clevercsv
Installing collected packages: chardet, regex, clevercsv
    Running setup.py install for clevercsv ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\lcnittl\playground\.venv\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\lcnittl\\AppData\\Local\\Temp\\pip-req-build-j5a594ud\\setup.py'"'"'; __file__='"'"'C:\\Users\\lcnittl\\AppData\\Local\\Temp\\pip-req-build-j5a594ud\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\lcnittl\AppData\Local\Temp\pip-record-prlevfmk\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\lcnittl\playground\.venv\include\site\python3.8\clevercsv'
         cwd: C:\Users\lcnittl\AppData\Local\Temp\pip-req-build-j5a594ud\
    Complete output (68 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.8
    creating build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\break_ties.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\consistency.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\cparser_util.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\detect.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\detect_pattern.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\detect_type.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\dialect.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\dict_read_write.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\escape.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\exceptions.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\normal_form.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\potential_dialects.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\read.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\utils.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\wrappers.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\write.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\_optional.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\__init__.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\__main__.py -> build\lib.win-amd64-3.8\clevercsv
    copying clevercsv\__version__.py -> build\lib.win-amd64-3.8\clevercsv
    creating build\lib.win-amd64-3.8\clevercsv\console
    copying clevercsv\console\application.py -> build\lib.win-amd64-3.8\clevercsv\console
    copying clevercsv\console\config.py -> build\lib.win-amd64-3.8\clevercsv\console
    copying clevercsv\console\__init__.py -> build\lib.win-amd64-3.8\clevercsv\console
    creating build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\code.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\detect.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\explore.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\standardize.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\view.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\_utils.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    copying clevercsv\console\commands\__init__.py -> build\lib.win-amd64-3.8\clevercsv\console\commands
    running egg_info
    writing clevercsv.egg-info\PKG-INFO
    writing dependency_links to clevercsv.egg-info\dependency_links.txt
    writing entry points to clevercsv.egg-info\entry_points.txt
    writing requirements to clevercsv.egg-info\requires.txt
    writing top-level names to clevercsv.egg-info\top_level.txt
    reading manifest file 'clevercsv.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'requirements.txt'
    warning: no files found matching '*' under directory 'bin'
    no previously-included directories found matching 'tests\test_integration'
    warning: no previously-included files found matching 'Makefile'
    warning: no previously-included files found matching '.gitignore'
    warning: no previously-included files found matching '.travis.yml'
    warning: no previously-included files found matching '.readthedocs.yml'
    warning: no previously-included files found matching 'make_release.py'
    warning: no previously-included files found matching 'cgrep'
    warning: no previously-included files found matching 'vgrep'
    no previously-included directories found matching 'notes'
    no previously-included directories found matching 'auxiliary'
    writing manifest file 'clevercsv.egg-info\SOURCES.txt'
    running build_ext
    building 'clevercsv.cparser' extension
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\src
    C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\lcnittl\playground\.venv\include -Ic:\python\python38\include -Ic:\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" /Tcsrc/cparser.c /Fobuild\temp.win-amd64-3.8\Release\src/cparser.obj
    cparser.c
    c:\python\python38\include\pyconfig.h(206): fatal error C1083: Cannot open include file: 'basetsd.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\lcnittl\playground\.venv\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\lcnittl\\AppData\\Local\\Temp\\pip-req-build-j5a594ud\\setup.py'"'"'; __file__='"'"'C:\\Users\\lcnittl\\AppData\\Local\\Temp\\pip-req-build-j5a594ud\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\lcnittl\AppData\Local\Temp\pip-record-prlevfmk\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\lcnittl\playground\.venv\include\site\python3.8\clevercsv' Check the logs for full command output.
(.venv) PS C:\Users\lcnittl\playground>
```

Would you have a (quick) solution for this?

Thanks,
Christian

--- a/README.md
+++ b/README.md
@@ -341,15 +341,10 @@ repository:
 
 ```yaml
 repos:
-  - repo: local
+  - repo: https://github.com/alan-turing-institute/CleverCSV-pre-commit
+    rev: v0.6.6   # or any later version
     hooks:
       - id: clevercsv-standardize
-        name: CleverCSV Standardize
-        entry: clevercsv standardize --in-place
-        language: python
-        language_version: python3
-        types: ["csv"]
-        additional_dependencies: ["clevercsv[full]"]
 ```
 
 Finally, run ``pre-commit install`` to set up the git hook. Pre-commit will 

--- a/README.md
+++ b/README.md
@@ -150,7 +150,8 @@ easier. We currently have the following helper functions:
   [stream_dicts](https://clevercsv.readthedocs.io/en/latest/source/clevercsv.html#clevercsv.wrappers.stream_dicts) 
   is also available.
 * [write_table](https://clevercsv.readthedocs.io/en/latest/source/clevercsv.html#clevercsv.wrappers.write_table): 
-  write a table (a list of lists) to a file using the RFC-4180 dialect.
+  write a table (a list of lists) to a file using the 
+  [RFC-4180](https://tools.ietf.org/html/rfc4180) dialect.
 
 Of course, you can also use the traditional way of loading a CSV file, as in 
 the Python CSV module:
@@ -309,7 +310,7 @@ CleverCSV has loaded the data into the variable: df
 #### Standardize
 
 Use the ``standardize`` command when you want to rewrite a file using the 
-RFC-4180 standard:
+[RFC-4180 standard](https://tools.ietf.org/html/rfc4180):
 
 ```text
 $ clevercsv standardize --output imdb_standard.csv imdb.csv

--- a/README.md
+++ b/README.md
@@ -41,7 +41,7 @@ import it.*
 
 ---
 
-*Contents:* <a href="#introduction"><b>Introduction</b></a> | <a href="#installation"><b>Installation</b></a> | <a href="#usage"><b>Usage</b></a> | <a href="#python-library">Python Library</a> | <a href="#command-line-tool">Command-Line Tool</a> | <a href="#contributing"><b>Contributing</b></a> | <a href="#notes"><b>Notes</b></a>
+*Contents:* <a href="#introduction"><b>Introduction</b></a> | <a href="#installation"><b>Installation</b></a> | <a href="#usage"><b>Usage</b></a> | <a href="#python-library">Python Library</a> | <a href="#command-line-tool">Command-Line Tool</a> | <a href="#version-control-integration">Version Control Integration</a> | <a href="#contributing"><b>Contributing</b></a> | <a href="#notes"><b>Notes</b></a>
 
 ---
 
@@ -328,6 +328,33 @@ before viewing or saving:
 ```text
 $ clevercsv view --transpose imdb.csv
 ```
+
+### Version Control Integration
+
+If you'd like to make sure that you never commit a messy (non-standard) CSV 
+file to your repository, you can install a 
+[pre-commit](https://pre-commit.com/) hook. First, install pre-commit using 
+the [installation instructions](https://pre-commit.com/#install). Next, add 
+the following configuration to the ``.pre-commit-config.yaml`` file in your 
+repository:
+
+```yaml
+repos:
+  - repo: local
+    hooks:
+      - id: clevercsv-standardize
+        name: CleverCSV Standardize
+        entry: clevercsv standardize --in-place
+        language: python
+        language_version: python3
+        types: ["csv"]
+        additional_dependencies: ["clevercsv[full]"]
+```
+
+Finally, run ``pre-commit install`` to set up the git hook. Pre-commit will 
+now use CleverCSV to standardize your CSV files following 
+[RFC-4180](https://tools.ietf.org/html/rfc4180) whenever you commit a CSV file 
+to your repository.
 
 ## Contributing
 

